tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/trace/ITmfTrace.java

```diff
@@ -273,6 +273,37 @@ public interface ITmfTrace extends ITmfEventProvider {
      */
     @NonNull Iterable<@NonNull IAnalysisModule> getAnalysisModules();
 
+
+    /**
+     * Add an analysis module. It will return an existing analysis module with
```

Comment:
It's not clear to me what is returned. Is it for 'replacing' a module with same id if one exists (unusual case) and just adding one if it does not exist (normal case) and then it returns null? If that's the case I would suggest a wording similar to Map.put():
the previous module with the same ID, or null if there was none.
That would imply that the previous module has been removed, this should be specified if it's the case.

Reply:
Looking at the implementation, this only affects added modules. So you could not replace a built-in module. Should this be checked in the code and fail if a built-in module with same id exists?

I would suggest then to modify the wording to 'the previously added module ...'.

Reply:
I'll update the java-doc. I'll also throw an exception if a built-in analysis module is added (replaced) or removed.